ios/Classes/TiNfcVTagTechnology.m

```diff
@@ -9,4 +9,701 @@
+  uint8_t requestFlags = [requestFlagsValue unsignedCharValue];
+
+  NSNumber *keyIdentifier = [[args firstObject] valueForKey:@"keyIdentifier"];
+
+  NSArray *message = [[args firstObject] valueForKey:@"message"];
+  unsigned int dataValue[message.count];
+  for (int i = 0; i < message.count; i++) {
+    dataValue[i] = [message[i] unsignedIntValue];
+  }
+  NSData *data = [[NSData alloc] initWithBytes:dataValue length:message.count];
+
+  [[self.tagProxy asNFCISO15693Tag] keyUpdateWithRequestFlags:requestFlags
+                                                keyIdentifier:[keyIdentifier integerValue]
+                                                      message:data
+                                            completionHandler:^(NFCISO15693ResponseFlag responseFlag, NSData *_Nonnull response, NSError *_Nullable error) {
+                                              if (![self _hasListeners:@"didKeyUpdateWithRequestFlags"]) {
+                                                return;
+                                              }
+                                              [self fireEvent:@"didKeyUpdateWithRequestFlags"
+                                                   withObject:@{
+                                                     @"errorCode" : error != nil ? NUMINTEGER([error code]) : [NSNull null],
+                                                     @"errorDescription" : error != nil ? [error localizedDescription] : [NSNull null],
+                                                     @"errorDomain" : error != nil ? [error domain] : [NSNull null]
+                                                   }];
+                                            }];
+}
+
+- (void)readBufferWithRequestFlags:(id)args
+{
+  NSNumber *requestFlagsValue = [[args firstObject] valueForKey:@"requestFlags"];
+  uint8_t requestFlags = [requestFlagsValue unsignedCharValue];
+
+  [[self.tagProxy asNFCISO15693Tag] readBufferWithRequestFlags:requestFlags
+                                             completionHandler:^(NFCISO15693ResponseFlag responseFlag, NSData *_Nonnull data, NSError *_Nullable error) {
+                                               if (![self _hasListeners:@"didReadBufferWithRequestFlags"]) {
+                                                 return;
+                                               }
+                                               [self fireEvent:@"didReadBufferWithRequestFlags"
+                                                    withObject:@{
+                                                      @"errorCode" : error != nil ? NUMINTEGER([error code]) : [NSNull null],
+                                                      @"errorDescription" : error != nil ? [error localizedDescription] : [NSNull null],
+                                                      @"errorDomain" : error != nil ? [error domain] : [NSNull null]
+                                                    }];
```

Comment:
Return other information like responseFlag and data with payload.

Reply:
Done